src/core/DataFile.cpp

```diff
@@ -1476,11 +1505,9 @@ void DataFile::loadData( const QByteArray & _data, const QString & _sourceFile )
 			}
 
 			// the upgrade needs to happen after the warning as it updates the project version.
-			if( createdWith.setCompareType( ProjectVersion::Build )
-								< openedWith )
-			{
-				upgrade();
-			}
+			if( createdWith.setCompareType( ProjectVersion::All ) <
+				openedWith.setCompareType( ProjectVersion::All ) )
```

Comment:
I find this is easier to read using a leading operator rather than trailing.
```suggestion
			if( createdWith.setCompareType( ProjectVersion::All )
				< openedWith.setCompareType( ProjectVersion::All ) )
```

Reply:
I made a somewhat drastic change to fix this that avoids these specific `setCompareType` calls completely, thus fitting the condition on one line. For the other long comparison I moved the comparator to the start of the second line as you suggested.